sdk/core/azure-core/src/main/java/com/azure/core/util/UserAgentUtil.java

```diff
@@ -12,6 +14,7 @@
     private static final String INVALID_APPLICATION_ID_LENGTH = "'applicationId' length cannot be greater than "
         + MAX_APPLICATION_ID_LENGTH;
     private static final String INVALID_APPLICATION_ID_SPACE = "'applicationId' cannot contain spaces.";
+    private static final Pattern AZURE_PREFIX = Pattern.compile("^azure-");
```

Comment:
Given that this is using `^` to begin the regex pattern we could instead use `startsWith` which should be more performant as there isn't the need to build all the internal structures that the `Matcher` needs.

Reply:
Simplified the removal of `azure-` prefix from the User Agent string. This should be executed just once per client creation, so, it won't be a perf bottleneck.